x-pack/plugins/viz_editor/public/editor_plugin_registry.ts

```diff
@@ -28,36 +32,34 @@ export interface Suggestion<S extends ViewModel = ViewModel> {
  *    Also contains a score which is used to sort the suggestions from all plugins
  */
 
-export interface EditorPlugin<S extends ViewModel = ViewModel> {
+export interface EditorPlugin<S extends VisModel = VisModel> {
   name: string;
   DataPanel: React.ComponentType<PanelComponentProps<S>>;
   ConfigPanel: React.ComponentType<PanelComponentProps<S>>;
   HeaderPanel?: React.ComponentType<PanelComponentProps<S>>;
-  TopAxisPanel?: React.ComponentType<PanelComponentProps<S>>;
-  BottomAxisPanel?: React.ComponentType<PanelComponentProps<S>>;
-  LeftAxisPanel?: React.ComponentType<PanelComponentProps<S>>;
-  RightAxisPanel?: React.ComponentType<PanelComponentProps<S>>;
-  toExpression: (viewModel: S, mode: 'view' | 'edit') => string;
-  getSuggestions: (viewModel: S) => Array<Suggestion<S>>;
-  getInitialState: (viewModel: S) => S;
+  WorkspacePanel?: React.ComponentType<PanelComponentProps<S>>;
+  toExpression: (visModel: S, mode: 'view' | 'edit') => string;
```

Comment:
I wonder if toExpression and getSuggestions could also be optional. In the case of the `toExpression`, we can probably have general function that will work for most visualizations: `esquery { query } | fooVisualization { config }`

Reply:
Good idea, I think it makes sense to pull as much as possible into the frame and common to keep the actual plugin definition as lean as possible. I will add placeholders for that. I'm not sure how much sense it makes for expressions though, but time will tell :)